src/cargo/util/paths.rs

```diff
@@ -87,6 +89,42 @@ pub fn write(path: &Path, contents: &[u8]) -> CargoResult<()> {
     })
 }
 
+pub fn append(path: &Path, contents: &[u8]) -> CargoResult<()> {
+    (|| -> CargoResult<()> {
+        let mut f = try!(
+                    OpenOptions::new()
+                        .write(true)
+                        .append(true)
+                        .create(true)
+                        .open(path));
+
+        try!(f.write_all(contents));
+        Ok(())
+    }).chain_error(|| {
+        internal(format!("failed to write `{}`", path.display()))
+    })
+}
+
+pub fn write_if_not_exists(path: &Path, contents: &[u8]) -> CargoResult<()> {
```

Comment:
This helper function seems a little dangerous as it doesn't allow distinguishing the case where a file existed or the file was successfully written, perhaps the relevant locations this is called could just have guards on `paths::is_file`?